CMakeLists.txt

```diff
@@ -64,8 +64,10 @@ endif()
 
 file(GLOB HEADERS "include/chainbase/*.hpp")
 add_library( chainbase src/chainbase.cpp src/pinnable_mapped_file.cpp ${HEADERS} )
-target_link_libraries( chainbase ${PLATFORM_LIBRARIES} )
-target_include_directories( chainbase PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" )
+target_link_libraries( chainbase ${PLATFORM_LIBRARIES} Boost::boost)
+target_include_directories( chainbase PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
```

Comment:
Does this change have any effect since we're not using `export()`/`install(EXPORT)`? If keeping this change I think you should change `${CMAKE_CURRENT_SOURCE_DIR}/include` to just simply `include` because while I'm not 100% sure, I think this change as-is may break builds if there are spaces. (the removed line had quotes around the path)

Reply:
I don't believe it has any effect when you don't use `export()`. I chose not to quote the generator expression simply because official cmake documentation don't use quote. To be honest, I don't exactly know what happen if there a space in ${CMAKE_CURRENT_SOURCE_DIR}, so I just added the quotes so it definitely won't cause problem.

Reply:
I just experimented with the case where ${CMAKE_CURRENT_SOURCE_DIR} has space and there's no quote surround the generator expression. It compiles without any issues.